src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -31,92 +31,73 @@ private BoundExpression VisitCompoundAssignmentOperator(BoundCompoundAssignmentO
-            // We rewrite dyn.Member op= RHS to the following:
-            //
-            //   IsEvent("Member", dyn) ? InvokeMember("{add|remove}_Member", dyn, RHS) : SetMember(BinaryOperation("op=", GetMember("Member", dyn)), RHS)
-            //
-            bool isPossibleEventHandlerOperation = node.Left.Kind == BoundKind.DynamicMemberAccess &&
-                (binaryOperator == BinaryOperatorKind.Addition || binaryOperator == BinaryOperatorKind.Subtraction);
-
             var lhsRead = MakeRValue(transformedLHS);
+            BoundExpression rewrittenAssignment;
 
-            // If RHS needs to be saved to a temp, we save the LHS read to a temp first to ensure order
-            // of operations is preserved
-            if (isPossibleEventHandlerOperation && CanChangeValueBetweenReads(loweredRight))
+            if (node.Left.Kind == BoundKind.DynamicMemberAccess &&
+                (binaryOperator == BinaryOperatorKind.Addition || binaryOperator == BinaryOperatorKind.Subtraction))
             {
-                lhsRead = storeToTemp(lhsRead);
-                loweredRight = storeToTemp(loweredRight);
-
-                BoundExpression storeToTemp(BoundExpression expression)
-                {
-                    var temp = _factory.StoreToTemp(expression, out BoundAssignmentOperator assignmentToTemp);
-                    stores.Add(assignmentToTemp);
-                    temps.Add(temp.LocalSymbol);
-                    return temp;
-                }
-            }
-
-            SyntaxNode syntax = node.Syntax;
-
-            // OK, we now have the temporary declarations, the temporary stores, and the transformed left hand side.
-            // We need to generate 
-            //
-            // xlhs = (FINAL)((LEFT)xlhs op rhs)
-            //
-            // And then wrap it up with the generated temporaries.
-            //
-            // (The right hand side has already been converted to the type expected by the operator.)
-
-            BoundExpression opLHS = isDynamic ? lhsRead : MakeConversionNode(
-                syntax: syntax,
-                rewrittenOperand: lhsRead,
-                conversion: node.LeftConversion,
-                rewrittenType: node.Operator.LeftType,
-                @checked: isChecked);
-
-            BoundExpression operand = MakeBinaryOperator(syntax, node.Operator.Kind, opLHS, loweredRight, node.Operator.ReturnType, node.Operator.Method, isCompoundAssignment: true);
+                // If this could be an event assignment at runtime, we need to rewrite to the following form:
+                // Original:
+                //   receiver.EV += handler
+                // Rewritten:
+                //   dynamic memberAccessReceiver = receiver;
+                //   bool isEvent = is_Event(memberAccessReceiver, "EV");
+                //   dynamic possibleEvent = !isEvent ? memberAccessReceiver.EV : null;
```

Comment:
>possibleEvent [](start = 29, length = 13)

This is confusingly named; `possibleEvent` is never an event.  It is only set (to something non-`null` when it is NOT an event.